backend/db.js

```diff
@@ -3,46 +3,190 @@
+    }
 
-// close open connection
-function close() {
-    if (client) {
-        client.close();
-        client = null;
+    static close() {
+        if (Database.client) {
+            Database.client.close();
+            Database.client = null;
+        }
     }
-}
 
+    static store_user(body, next) {
+        return Database.db('GrandValet').collection('Users').findOne({username: body.username})
+        .then((user) => {
+            if (user == null) 
+            {
+                let newData = {type: body.type, username: body.username, name: body.name, password: body.password, email: body.email, phone: body.phone, driverStatus: body.driverStatus};
+                Database.db('GrandValet').collection('Users').insertOne(newData);
+                return;
+            }
+            Database.db('GrandValet').collection('Users').updateOne({username: body.username}, { $set: {type: body.type, name: body.name, password: body.password, email: body.email, phone: body.phone, driverStatus: body.driverStatus}});
+        });
+    }
+
+    static read_user(username, next) {
+        return Database.db('GrandValet').collection('Users').findOne({username: username})
+        .then((user) => {
+            if (user == null) 
+            {
+                console.log("bad")
+                return;
+            }
+            return user;
+        });
+    }
+    
+    static read_activeHubs(next) {
+        let time = Math.floor(Date.now() / 1000);
+        return Database.db('GrandValet').collection('Hubs').find({$and: [{ startTime : { $lt :  time}}, { endTime : { $gt :  time}}]}).toArray()
+        .then((hubs) => {
+            if (hubs == null) 
+            {
+                console.log("bad")
+                return;
+            }
+            return hubs;
+        });
+    }
+
+    // Radius is in miles, longitude first (this is important, as normally latitutde first)
+    static read_nearbyHubs(loc, radius, next) {
+        let milesToRadian = function(miles){
+            let earthRadiusInMiles = 3959;
+            return miles / earthRadiusInMiles;
+        };
+        let query = {
+            "location" : {
+                $geoWithin : {
+                    $centerSphere : [loc, milesToRadian(radius) ]
+                }
+            }
+        };
+        return Database.db('GrandValet').collection('Hubs').find(query).toArray()
+        .then((hubs) => {
+            if (hubs == null) 
+            {
+                console.log("bad")
+                return;
+            }
+            return hubs;
+        });
+    }
+
+    static read_hub(hubId, next) {
+        return Database.db('GrandValet').collection('Hubs').findOne({hubId: hubId})
+        .then((hubs) => {
+            if (hubs == null) 
+            {
+                console.log("bad")
+                return;
+            }
+            return hubs;
+        });
+    }
+
+    static store_hub(body, next) {
+        console.log(body);

```

Comment:
If you are confident in the implementation, I think we can remove the logging here. Or maybe we can add some indications to prevent confusing this message with other debugging messages

Reply:
Solved.